fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidVariant.java

```diff
@@ -84,4 +84,11 @@ default Fluid getFluid() {
 	default RegistryEntry<Fluid> getRegistryEntry() {
 		return getFluid().getRegistryEntry();
 	}
+
+	/**
+	 * Creates a copy of this FluidVariant with the provided component changes applied.
+	 * @param changes the changes to apply
+	 * @return the new variant with the changes applied
+	 */
+	FluidVariant withChanges(ComponentChanges changes);
```

Comment:
`withComponentChanges` is a better name to clarify what we're changing IMO

Reply:
:+1: